.editorconfig

```diff
@@ -0,0 +1,14 @@
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
```

Comment:
in theory i have no qualms with an editorconfig, I know that it's becoming fairly common.  But in this repo (and all of our typescript repos) the linting rules are expressed via `eslint` and `prettier`.  Those rules are enforced in CI, recognized by many (most?) editors, and applied via `npm run format`.

So I am concerned about whether having this file here might cause competing opinions where things start getting flipped back and forth by editorconfig battling with eslint/prettier.  And also expressing these rules in 2 places seems like it might require extra care to make sure we keep them in sync.

Maybe you have experience with using both editorconfig and eslint/prettier before?  If so, and my concerns are unfounded, lmk!

Reply:
On the contrary, I added it because the linters complained about trailing whitespace.

Think of it this way. EditorConfig does not define the style, it helps editors follow the style. And the style encoded in EditorConfig is so basic the redundancy doesn't matter.

Reply:
okay.  i am willing to trust your judgment on it.  FWIW I have never configured my editor at all, it just understands the eslint and prettier settings; hence my concern that editorconfig might confuse it rather than help.  But you and I are using different editors so I'm fine with trying this.

Reply:
If it confuses editors, we can remove it.

Reply:
agreed!